src-backbone/app/js/controllers/conceptsController.js

```diff
@@ -0,0 +1,27 @@
+const Config = require('utils/config');
+
+const App                  = require('utils/sanaAppInstance');
+const Helpers              = require('utils/helpers');
+const ConceptsLayout       = require('views/concepts/conceptsLayoutView');
+const ConceptsUploaderView = require('views/concepts/conceptsUploaderView');
+
+module.exports = Marionette.Controller.extend({
+
+    routeUploader: function () {
+        if (!App().session.isValid()) {
+            Helpers.navigateToDefaultLoggedOut();
+            return;
+        }
+        App().session.refreshUser(false);
```

Comment:
You're freezing the UI thread here. Why not just let the async callback redirect the user instead?